### ISSUE TITLE
Flaky test for single pod integration

### ISSUE THREAD

**Issue by alculquicondor** [kubernetes-sigs/kueue#1559]
**What happened**:

```
Expected pod to be deleted
The function passed to Eventually failed at /home/prow/go/src/sigs.k8s.io/kueue/test/integration/controller/jobs/pod/pod_controller_test.go:297 with:
Expected success, but got an error:
    <*errors.StatusError | 0xc00102f5e0>: 
    pods "test-pod" not found
    {
        ErrStatus: {
            TypeMeta: {Kind: "", APIVersion: ""},
            ListMeta: {
                SelfLink: "",
                ResourceVersion: "",
                Continue: "",
                RemainingItemCount: nil,
            },
            Status: "Failure",
            Message: "pods \"test-pod\" not found",
            Reason: "NotFound",
            Details: {Name: "test-pod", Group: "", Kind: "pods", UID: "", Causes: nil, RetryAfterSeconds: 0},
            Code: 404,
        },
    }
```

**What you expected to happen**:

It looks like the Pod was actually deleted, but somehow the check is wrong.

**How to reproduce it (as minimally and precisely as possible)**:

https://testgrid.k8s.io/sig-scheduling#periodic-kueue-test-integration-main&width=20

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
- Kueue version (use `git describe --tags --dirty --always`):
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

**Comment by alculquicondor** [comment]
cc @achernevskii
/assign

**Comment by tenzen-y** [comment]
@alculquicondor Isn't this duplicated with #1501?

**Comment by tenzen-y** [comment]
And then, I fixed it in #1547

**Comment by alculquicondor** [comment]
Oops, sorry, I forgot to check for existing issues

/close
